src/libstd/io/stdio.rs

```diff
@@ -872,6 +874,82 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Equivalent to the [`input`] method except that a prompt
+/// is printed before capturing input. This function returns
+/// either success ([`Ok`]) or failure ([`Err`]) containing
+/// the input. See the [`io::Result`] module documentation
+/// for details.
+///
+/// # Note
+///
+/// If you require more explicit control over capturing
+/// user input, see the [`Stdin::read_line`] method.
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(prompt)]
+/// use std::io::prompt;
+///
+/// fn main() {
+///     let name = prompt("Enter name: ").expect("input failed!");
+///
+///     println!("Your name is {}!", name);
+/// }
```

Comment:
```suggestion
/// let name = prompt("Enter name: ").expect("input failed");
/// println!("Your name is {}!", name);
```
I think we can just keep this short since this is just 2 lines. Hopefully, beginners know that `main` is required when they reached here since `cargo new` did that for them.